integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/util/SourceControlUtils.java

```diff
@@ -0,0 +1,282 @@
+        String payloadString = payload.toString();
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put(HttpHeaders.AUTHORIZATION, "Bearer " + accessToken);
+        headers.put("Content-Type", "application/json");
+
+        HttpResponse response = HttpClientRequest.doPost(GIT_API_URL + "/projects/" + GIT_USERNAME + "%2F" + GIT_PROJECT_PATH + "/repository/commits", payloadString, headers);
+
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_CREATED);
+
+        log.info(commitMessage + " : Artifacts committed to Git repository successfully");
+    }
+
+    /**
+     * Reads the given directory
+     *
+     * @param directory                Path of the directory
+     * @param baseDirectory            Path of the base directory (artifacts directory)
+     * @param actions                  JSON Array of actions for committing the files to the repository
+     * @throws FileNotFoundException   If an error occurs when reading the directory
+     */
+    public static void readArtifactsDirectory(File directory, String baseDirectory, JSONArray actions, Map<String, String> fileActions) throws Exception {
+        List<String> filePaths = new ArrayList<>();
+        getFiles(directory, filePaths);
+        for (String filePath : filePaths){
+            // Convert the filePath to a relative path
+            String relativePath = filePath.replace(baseDirectory, ".");
+            if (fileActions.containsKey(filePath)){
+                String fileAction = fileActions.get(filePath);
+                if (fileAction.equals(ADD_FILE)){
+                    File addedFile = new File(filePath);
+                    JSONObject action = addFile(addedFile, relativePath);
+                    actions.put(action);
+                } else if (fileAction.equals(UPDATE_FILE)){
+                    File updatedFile = new File(filePath);
+                    JSONObject action = updateFile(updatedFile, relativePath);
+                    actions.put(action);
+                } else if (fileAction.equals(DELETE_FILE)){
+                    JSONObject action = deleteFile(relativePath);
+                    actions.put(action);
+                }
+            }
+        }
+    }
+
+    /**
+     * Reads the given directory and adds the file paths to the given list
+     *
+     * @param directory               Path of the directory
+     * @param filesList               List of file paths
+     */
+    public static void getFiles(File directory, List<String> filesList){
+        File[] files = directory.listFiles();
+        for (File file : files){
+            if (file.isDirectory()){
+                getFiles(file, filesList);
+            } else {
+                String filePath = file.getPath();
+                filesList.add(filePath);
+            }
+        }
+    }
+
+    /**
+     * Reads the content of the given file
+     *
+     * @param file
```

Comment:
Shall we complete explanations relevant to the comment ? 
Please check below methods in this file also. (method parameter descriptions)